src/view/writer.js

```diff
@@ -12,704 +12,993 @@ import ContainerElement from './containerelement';
-		 *
-		 * @error view-writer-break-root
-		 */
-		throw new CKEditorError( 'view-writer-break-root: Trying to break root element.' );
+	/**
+	 * Moves {@link module:engine/view/selection~Selection#focus selection's focus} to the specified location.
+	 *
+	 * The location can be specified in the same form as {@link module:engine/view/position~Position.createAt} parameters.
+	 *
+	 * @param {module:engine/view/item~Item|module:engine/view/position~Position} itemOrPosition
+	 * @param {Number|'end'|'before'|'after'} [offset=0] Offset or one of the flags. Used only when
+	 * first parameter is a {@link module:engine/view/item~Item view item}.
+	 */
+	setSelectionFocus( itemOrPosition, offset ) {
+		this.document.selection._setFocus( itemOrPosition, offset );
 	}
 
-	if ( position.isAtStart ) {
-		return Position.createBefore( element );
-	} else if ( !position.isAtEnd ) {
-		const newElement = element.clone( false );
+	/**
+	 * Sets {@link module:engine/view/selection~Selection selection's} to be marked as `fake`. A fake selection does
+	 * not render as browser native selection over selected elements and is hidden to the user.
+	 * This way, no native selection UI artifacts are displayed to the user and selection over elements can be
+	 * represented in other way, for example by applying proper CSS class.
+	 *
+	 * Additionally fake's selection label can be provided. It will be used to describe fake selection in DOM (and be
+	 * properly handled by screen readers).
+	 *
+	 * @param {Boolean} [value=true] If set to true selection will be marked as `fake`.
+	 * @param {Object} [options] Additional options.
+	 * @param {String} [options.label=''] Fake selection label.
+	 */
+	setFakeSelection( value, options ) {
+		this.document.selection._setFake( value, options );
+	}
 
-		insert( Position.createAfter( element ), newElement );
+	/**
+	 * Creates a new {@link module:engine/view/text~Text text node}.
+	 *
+	 *		writer.createText( 'foo' );
+	 *
+	 * @param {String} data Text data.
+	 * @returns {module:engine/view/text~Text} Created text node.
+	 */
+	createText( data ) {
+		return new Text( data );
+	}
 
-		const sourceRange = new Range( position, Position.createAt( element, 'end' ) );
-		const targetPosition = new Position( newElement, 0 );
+	/**
+	 * Creates new {@link module:engine/view/element~Element element}.
+	 *
+	 *		writer.createElement( 'paragraph' );
+	 *		writer.createElement( 'paragraph', { 'alignment': 'center' } );
+	 *
+	 * @param {String} name Name of the element.
+	 * @param {Object} [attributes] Elements attributes.
+	 * @returns {module:engine/view/element~Element} Created element.
+	 */
+	createElement( name, attributes ) {
+		return new Element( name, attributes );
+	}
 
-		move( sourceRange, targetPosition );
+	/**
+	 * Creates new {@link module:engine/view/attributeelement~AttributeElement}.
+	 *
+	 *		writer.createAttributeElement( 'paragraph' );
+	 *		writer.createAttributeElement( 'paragraph', { 'alignment': 'center' } );
```

Comment:
Paragraph is not an attribute element, so the example is wrong. You can use `strong` here.